src/analyzer/analyzerthread.cpp

```diff
@@ -204,14 +204,14 @@ bool AnalyzerThread::submitNextTrack(TrackPointer nextTrack) {
 }
 
 WorkerThread::TryFetchWorkItemsResult AnalyzerThread::tryFetchWorkItems() {
-    DEBUG_ASSERT(!m_currentTrack);
-    TrackPointer* pFront = m_nextTrack.front();
+    DEBUG_ASSERT(!m_currentTrack.has_value());
+    AnalyzerTrack* pFront = m_nextTrack.front();
     if (pFront) {
-        m_currentTrack = *pFront;
+        *m_currentTrack = *pFront;
```

Comment:
I'm getting segfaults here during the decrement of the `TrackPointer`'s reference count (inside the `AnalyzerTrack` that is getting replaced) and I'm not really sure why, since I don't see where the `AnalyzerTrack` is handled unsafely. Perhaps someone else has an idea what the issue could be?